specs/market-api.yaml

```diff
@@ -528,18 +494,16 @@ paths:
               #operationRef: '#/paths/~1agreements~1{agreementId}/get'
               parameters:
                 agreementId: '$response.body'
-            # TODO: more links
-        '400':
-          $ref: '#/components/responses/BadRequest'
-        '401':
-          $ref: '#/components/responses/Unauthorized'
-        '409':
+        400:
+          $ref: 'common.yaml#/responses/BadRequest'
+        401:
+          $ref: 'common.yaml#/responses/Unauthorized'
+        409:
           description: Proposal not in `Draft` state.
-        '410':
-          $ref: '#/components/responses/Expired'
+        410:
+          $ref: 'common.yaml#/responses/Gone'
```

Comment:
I believe freshly Agreement can't be `Gone`. I was considering if `Proposal` being promoted to `Agreement` can be expired, but `Proposal` has no validity period, so I think `410` here simply cannot happen.

Reply:
Hmm, we have `expiration_ts` internally for Proposal, but until we introduce it on `yaml` level we shouldn't expect `410` here

Reply:
Hmm, but Proposal State can be Expired, so we should include `validTo` field there to be consistent.

Reply:
We'd decided that `400 BadRequest` will handle Expired/Rejected Proposal case

Reply:
Agreed.

Reply:
After more indepth consideration I've decided to use `409  Conflict` for the Proposal being in `Initial` state, because it is not permanent failure (next invocation of `createAgreement` may still succeed) and `410 Gone` for  the Proposal being rejected or expired, because it is permanent (next invocations of `createAgreement` will always fail).

Pls note that we allow multiple `createAgreement` invocations for the same Proposal (ie. first moves from `Draft` to the `Accepted` state, and consecutive do not change `Proposal` state, it is still `Accepted`).